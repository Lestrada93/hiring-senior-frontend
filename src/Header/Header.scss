.navbar {
  margin-bottom: 15px;
  &--toggle {
    display: block !important;
    transition: all 0.5s linear;
  }
}