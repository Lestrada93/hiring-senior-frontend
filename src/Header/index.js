import React from "react";
import { Link } from "react-router-dom";
import "./Header.scss";
import PropTypes from "prop-types";
import {withTranslation} from "react-i18next";

/**
 * Header
 * @description Header component
 * @param t
 * @return {JSX.Element}
 * @constructor
 */
function Header({t}){

    const handleClick = () => {
        const navBar = document.getElementById("navbarNavAltMarkup");
        navBar.classList.toggle('navbar--toggle');
    }

    return (
        <header>
            <nav className="navbar navbar-expand-lg navbar-light bg-light">
                <div className="container-fluid">
                    <Link to="/" className="navbar-brand">
                        {t('VIA')}
                    </Link>
                    <button className="navbar-toggler" type="button" data-bs-toggle="collapse"
                            data-bs-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup"
                            aria-expanded="false" aria-label="Toggle navigation" onClick={handleClick}>
                        <span className="navbar-toggler-icon"></span>
                    </button>
                    <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                        <div className="navbar-nav">
                            <Link to="/" exact className="nav-link">
                                {t('LIST_OF_INVOICES')}
                            </Link>
                            <Link to="/create-a-new-invoice" className="nav-link">
                                {t('CREATE_A_NEW_INVOICE')}
                            </Link>
                        </div>
                    </div>
                </div>
            </nav>
        </header>
    );
}

Header.propTypes = {
    t: PropTypes.func,
};

Header.defaultProps = {
    t: () => {},
};

export default withTranslation()(Header);