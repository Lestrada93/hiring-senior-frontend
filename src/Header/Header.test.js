import React from 'react';
import { create } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '.';

describe('Header component', () => {
    test('Snapshot', () => {
        const header = create(
            <Router>
                <Header />
            </Router>);
        expect(header.toJSON()).toMatchSnapshot();
    });
});
