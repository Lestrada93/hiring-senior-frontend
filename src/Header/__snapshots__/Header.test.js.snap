// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Snapshot 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        VIA
      </a>
      <button
        aria-controls="navbarNavAltMarkup"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarNavAltMarkup"
        data-bs-toggle="collapse"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            className="nav-link"
            exact={true}
            href="/"
            onClick={[Function]}
          >
            LIST_OF_INVOICES
          </a>
          <a
            className="nav-link"
            href="/create-a-new-invoice"
            onClick={[Function]}
          >
            CREATE_A_NEW_INVOICE
          </a>
        </div>
      </div>
    </div>
  </nav>
</header>
`;
